lib/secretsscanner/authorizedkeys/authorized_keys.go

```diff
@@ -272,6 +296,38 @@ func (w *Watcher) fetchAndReportAuthorizedKeys(
 			keys = append(keys, hostKeys...)
 		}
 	}
+	return keys, nil
+}
+
+// errKeysNotReported is returned when the keys are not reported.
+var errKeysNotReported = errors.New("keys not reported")
+
+// fetchAndReportAuthorizedKeys fetches the authorized keys from the system and reports them to the cluster.
+func (w *Watcher) fetchAndReportAuthorizedKeys(
+	ctx context.Context,
+	fileWatcher *fsnotify.Watcher,
+	requiresReportToExtendTTL bool,
+) (returnErr error) {
```

Comment:
This could return `(bool, error)` where `bool` is whether or not the keys were reported. That way the `errKeysNotReported` is no longer needed, since it's not really an error, and the logic is more explicit about why the expiration timer is being reset. Kind of a nitpick though.

Reply:
handled in [de5c695](https://github.com/gravitational/teleport/pull/48627/commits/de5c6957af25fc2592cbdb6413e1049a73fb3bc9)